### ISSUE TITLE
IMPORT PGDUMP docs issues

### ISSUE THREAD

**Issue by rytaft** [cockroachdb/docs#3577]
While doing QA for https://github.com/cockroachdb/cockroach/issues/28158, I ran into a few issues with the docs that may (or may not) be worth fixing. Feel free to split this into multiple issues.

I was working from the documentation here for `IMPORT PGDUMP`: https://www.cockroachlabs.com/docs/dev/import-data.html.

1. I tried to follow the example with the `employees` database, but ran into difficulty with `pgloader`. I was able to download the dataset and load up MySQL, but `pgloader` failed. Eventually I ran `mysqldump`, and then manually edited the file until I could load it into Postgres using the `\i` Postgres CLI command. It seems that `pgloader` doesn't play nice with MySQL 8.0. Perhaps there is another database we can suggest that can be loaded directly into Postgres? 

2. I tried to follow the suggestion in the docs to use an http server to serve my files, but I couldn't figure out how. The command line example in the docs for uploading with caddy did not work. I think [this documentation](https://www.cockroachlabs.com/docs/dev/create-a-file-server.html#using-caddy-as-a-file-server) could probably be expanded to help users not familiar with how it works.

 3. Since I couldn't get the http server to work, I decided to use my local machine. Where to put the data on my local machine was a bit confusing at first. I didn't realize it had to be in `cockroach-data/extern`. The docs just say "an extern subdirectory", but don't specify exactly where.

 4. Finally I was able to run the example `IMPORT` command in the `IMPORT PGDUMP` docs, and loaded the data successfully. The docs example showing success is confusing, though, because it shows 0 rows and 0 bytes loaded. In reality, this should be a positive number.
Docs example:
  ```
  +--------------------+-----------+--------------------+------+---------------+----------------+-------+
  |       job_id       |  status   | fraction_completed | rows | index_entries | system_records | bytes |
  +--------------------+-----------+--------------------+------+---------------+----------------+-------+
  | 352938237301293057 | succeeded |                  1 |    0 |             0 |              0 |     0 |
  +--------------------+-----------+--------------------+------+---------------+----------------+-------+
  (1 row)
  ```
  What it really looks like:
  ```
          job_id       |  status   | fraction_completed |  rows  | index_entries | system_records |  bytes    
  +--------------------+-----------+--------------------+--------+---------------+----------------+----------+
    377028113937956865 | succeeded |                  1 | 300024 |             0 |              0 | 11534293  
  (1 row)
  ```

5. I tried removing all the flags that docs said were required (`--inserts`, `--no-privileges` and `--no-owner`), and it actually worked fine... `COPY` works successfully, and CockroachDB ignores the statements related to privileges and owners. It might also help to explicitly say that the `-t` flag shown in the docs example is optional. It should still work if you dump the entire database, not just one table.

6. The import fails when the pg_dump file contains functions, triggers, or type definitions. It might be good to have documentation for that. The `--disable-triggers` flag can be used to ensure triggers are omitted from the pg_dump file. 

7. Some pg_dump flags that don't work at all with CockroachDB's `IMPORT PGDUMP` (but a user might think they should) are `-a/--data-only`, `-c/--clean`,  and `-C/--create`.

8. There are two other `IMPORT PGDUMP` variants that the docs don't mention yet. As described by @dt they are:
  - `IMPORT PGDUMP 'pg_dump_file';` Run this with any pg_dump file and it should Just Work™.
  - `IMPORT TABLE foo FROM PGDUMP 'pg_dump_file';` Run this with any pg_dump file and it should just import "foo" even if the dump contains other things too.

**Comment by jseldess** [comment]
Thanks very much for this thorough feedback, @rytaft! @rmloveland is working on these docs, so I'm assigning this to him.

**Comment by rmloveland** [comment]
@rytaft I echo Jesse's thanks for this very thorough feedback!

I have just submitted a PR (#3766) which I believe addresses the issues you've listed, but to make sure everything is captured I wrote responses to each below.  Thank you again for this very helpful issue!

> 1. I tried to follow the example with the employees database, but ran into difficulty with pgloader.

To avoid this hitting end users I did the conversion locally and put the resulting (mysqldump, pgdump, csv) files in S3.  The various IMPORT examples in the PR linked above (e.g. [Migrate from Postgres](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/216350d714edcb40bcdd2eff7d94d1b031ac4bef/dev/migrate-from-postgres.html) ) then pull these files down directly.

> 2. I tried to follow the suggestion in the docs to use an http server to serve my files, but I couldn't figure out how.

I updated the [file server page](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/216350d714edcb40bcdd2eff7d94d1b031ac4bef/dev/create-a-file-server.html#using-python-with-import)  with some PHP, Python, and Ruby one-liners that worked for local IMPORT runs.  All of which are much easier to use (IMO) than caddy, which I find confusing in general, and nginx which is rather heavy for quick one-off things.

> 3. Since I couldn't get the http server to work, I decided to use my local machine. Where to put the data on my local machine was a bit confusing at first.

Agreed!  Updated the [import file location](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/216350d714edcb40bcdd2eff7d94d1b031ac4bef/dev/import.html#import-file-location) section to (hopefully) more clearly describe how the local dir option works (but it may be less necessary given that a one-liner Python web server method is doc'd, so IDK)

> 4. Finally I was able to run the example IMPORT command in the IMPORT PGDUMP docs, and loaded the data successfully. The docs example showing success is confusing, though, because it shows 0 rows and 0 bytes loaded.

All uses of `IMPORT` in the PR mentioned above have been updated and I don't see anything with 0 bytes now (though I may have missed something, off to double check).  Sorry for the confusion, not sure how that happened.

> 5. I tried removing all the flags that docs said were required (--inserts, --no-privileges and --no-owner), and it actually worked fine... COPY works successfully, and CockroachDB ignores the statements related to privileges and owners.

This (dropping all the old flags) also worked for me, except in one case where I tried pulling in a full `pg_dump` and needed to add `--no-privileges` in order to keep it from blowing up on  a line with `GRANT ALL ON SCHEMA public TO PUBLIC` but I am planning to raise this with reviewers and correct it as needed on the PR discussion, since I suspect I am doing something wrong: I don't really understand PG schemas and just cargo-culted that flag back in to make the import work for this one database.

>  It might also help to explicitly say that the -t flag shown in the docs example is optional. It should still work if you dump the entire database, not just one table.

Agreed, I think this is addressed by splitting out the "dump the entire DB" and "dump a table" cases in the (proposed) new [Migrate from Postgres](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/216350d714edcb40bcdd2eff7d94d1b031ac4bef/dev/migrate-from-postgres.html#step-1-dump-the-postgres-database) page.

> 6. The import fails when the pg_dump file contains functions, triggers, or type definitions. It might be good to have documentation for that. The --disable-triggers flag can be used to ensure triggers are omitted from the pg_dump file.

Added to the 'Migrate from Postgres' page linked above, thanks.

> 7. Some pg_dump flags that don't work at all with CockroachDB's IMPORT PGDUMP (but a user might think they should) are -a/--data-only, -c/--clean, and -C/--create.

I did not add these to the linked PR, with the reasoning that the docs should say exactly what the user should do rather than what they should not, since `pg_dump` has many flags.  However, that may be wrong.  Would it also work to say something like "don't add any other flags to `pg_dump`, these are the known working flags"?

> 8. There are two other IMPORT PGDUMP variants that the docs don't mention yet. 

Added these to the 'Migrate to Postgres' page in the linked PR.  Thank you!
